src/App.tsx

```diff
@@ -14,11 +14,66 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          {!selectedGood.length
+            ? 'No goods selected'
+            : `${selectedGood}
+            ${' '}
+            is selected`}
+          {selectedGood.length
+            ? (
+              <button
+                type="button"
+                onClick={() => (
+                  this.setState({ selectedGood: [] })
+                )}
+              >
+                Clear
+              </button>
+            )
+            : null}
+        </h1>
+        <ul>
+          {goodsFromServer.map((good: string) => (
+            <li
+              key={good}
+              className={`${selectedGood.includes(good)
+                ? 'selectedGood'
+                : ''
+              }`}
+            >
+              {good}
+              {' '}
+              <button
+                type="button"
+                onClick={() => {
+                  this.setState({
+                    selectedGood: selectedGood.includes(good)
+                      ? selectedGood.filter((select: string) => select !== good)
+                      : [...selectedGood, good],
```

Comment:
Should be moved to a separate hander function

Reply:
I am sorry, something happened to my commit probably